test/integration/042_sources_tests/test_source_fresher_state.py

```diff
@@ -0,0 +1,685 @@
+import json
+import os
+from datetime import datetime, timedelta
+
+import yaml
+
+from dbt.exceptions import ParsingException
+import dbt.tracking
+import dbt.version
+from test.integration.base import DBTIntegrationTest, use_profile, AnyFloat, \
+    AnyStringWith
+
+
+class BaseSourcesTest(DBTIntegrationTest):
```

Comment:
We've _just_ converted `042_sources_tests` to use the new `pytest`-based framework (https://github.com/dbt-labs/dbt-core/pull/4935).

I'd say you have two valid options:
- Try rewriting these tests to use the new framework — I think it's a lot nicer! perhaps not all that different, for what you're trying to do here
- Pop this test over to `062_defer_state_tests`, which hasn't yet been converted (and is arguably closer kin for this functionality, anyway)

Reply:
We'll use the new pytest framework. It's worth learning upfront for many more PRs to come 😉